src/components/Login/Login.component.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import Styled from './Login.styled';
+import { useData } from '../../states/provider';
+import { ACTIONS } from '../../states/reducer';
+
+function ModalLogin({ show }) {
+    const { data, dispatch } = useData();
+    const [userName, setUserName] = useState(undefined);
```

Comment:
Leaving the `useState()` blank without any param is the save as saying `useState(undefined)`. What I do in the case of an input is initializing the state to an empty string like `useState('')`.